teacher_class.rb

```diff
@@ -0,0 +1,23 @@
+require_relative 'person'
+
+class Teacher < Person
+  def initialize(specialization, age, name = 'unknown', parent_permission: true)
+    super(name, age, parent_permission)
```

Comment:
- [ ] Define the parent_permission properly in the super class, based on the feedback above

- [ ] Also re-arrange your arguments here.

  ```
  super(age, name, parent_permission: parent_permission)
  ```